### PR TITLE
Separate message for class attributes from assignment

### PR DESCRIPTION
**Pull request by karamanolev**
This enables more precise configuration to enable/disable desired messages.

### CODE REVIEW

**Comment by danpalmer** [comment]
This looks great, would love to see this merged soon as we're also seeing all the issues mentioned in https://github.com/gforcada/flake8-builtins/issues/3

**Comment by karamanolev** [comment]
@danpalmer Hopefully, this gets reviewed soon, but until then, if you don't want to wait and you haven't already done it you can do like we did in our project and add `-e git+https://github.com/karamanolev/flake8-builtins.git@a5e39af6a335e9773865734d2750f8a58d469d64#egg=flake8-builtins` to your requirements.txt files.

**Comment by danpalmer** [comment]
@karamanolev thanks! I've already got my own fork with this and the other open PR merged in.

File: flake8_builtins.py [hunk 1]
```diff
@@ -64,16 +69,19 @@ def run(self):
                     yield line, offset, msg, rtype
 
     def check_assignment(self, statement):
+        is_class_def = type(statement.__flake8_builtins_parent) is ast.ClassDef
```
Comment (gforcada) [inline flake8_builtins.py]: I very much prefer to have a regular ``if`` conditional, these inline boolean expressions are easy to oversee while refactoring.

Something like:
```python
var = False
if condition:
  var = True
```

File: flake8_builtins.py [hunk 2]
```diff
@@ -64,16 +69,19 @@ def run(self):
...
+
         for element in statement.targets:
             if isinstance(element, ast.Name) and \
                     element.id in BUILTINS:
 
                 line = element.lineno
                 offset = element.col_offset
+                msg = self.class_attribute_msg if is_class_def else self.assign_msg
```
Comment (gforcada) [inline flake8_builtins.py]: Same as above, use an ``if`` please

File: run_tests.py [hunk 3]
```diff
@@ -1,17 +1,21 @@
 # -*- coding: utf-8 -*-
-from flake8_builtins import BuiltinsChecker
-
 import ast
 import unittest
 
+from flake8_builtins import BuiltinsChecker
+
```
Comment (gforcada) [inline run_tests.py]: Please revert this change and the one above, i.e. keep the imports as they were.

File: run_tests.py [hunk 4]
```diff
@@ -1,17 +1,21 @@
...
 
 class TestBuiltins(unittest.TestCase):
+    def assert_codes(self, ret, codes):
+        self.assertEqual(len(ret), len(codes))
+        for item, code in zip(ret, codes):
+            self.assertTrue(item[2].startswith(code + ' '))
```
Comment (gforcada) [inline run_tests.py:14]: That's nice! ❤️

**Review by gforcada — CHANGES_REQUESTED** [review]
@karamanolev thanks for providing the patch!! The code looks fine and you even added tests, awesome!

There are only some cosmetic changes I would like to have before merging.

I was on vacations, so I could not check it before.

Looking forward to merge it and release it soon 😉

File: run_tests.py (hunk 3 above)
Comment (karamanolev) [reply run_tests.py]: I'll do it, because it's your library after all, just please note that this will be breaking PEP8 in this section:

Imports are always put at the top of the file, just after any module comments and docstrings, and before module globals and constants.

Imports should be grouped in the following order:

standard library imports
related third party imports
local application/library specific imports
You should put a blank line between each group of imports.

**Comment by karamanolev** [comment]
@gforcada I think your changes are implemented. Please review.

File: run_tests.py (hunk 3 above)
Comment (gforcada) [reply run_tests.py]: @karamanolev yes, I'm aware of it, though, I'm usually working with Plone related code bases, and the Plone community has a different styleguide regarding imports: *sort them alphabetically*.

The use cases are different but I find it pleasantly to not have to think at all about imports, just sort them alphabetically (or let it isort do it for you) and carry on.

**Comment by gforcada** [comment]
@karamanolev thanks for the changes, I released it as 0.4, please give it a try and report back!
